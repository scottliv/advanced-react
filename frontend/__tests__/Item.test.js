import Item from "../components/Item";
import { shallow } from "enzyme";
import { wrap } from "module";
import toJson from "enzyme-to-json";

const fakeItem = {
  id: "4321",
  title: "An Item",
  description: "lookin good!",
  price: 350,
  image: "dog.jpg",
  largeImage: "bigDog.jpg"
};

describe("<Item/>", () => {
  it("renders and matches the snapshot", () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
