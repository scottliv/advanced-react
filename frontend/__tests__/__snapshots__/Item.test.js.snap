// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="An Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "4321",
          },
        }
      }
    >
      <a>
        An Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $3.50
  </PriceTag>
  <p>
    lookin good!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "4321",
          },
        }
      }
    >
      <a>
        Edit 
      </a>
    </Link>
    <AddToCart
      id="4321"
    />
    <DeleteItem
      id="4321"
    >
      Delete
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
